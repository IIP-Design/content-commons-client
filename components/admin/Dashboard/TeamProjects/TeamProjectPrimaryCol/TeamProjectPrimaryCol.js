/**
 *
 * TeamProjectPrimaryCol
 *
 */
import React, { Fragment, useContext } from 'react';
import PropTypes from 'prop-types';
import dynamic from 'next/dynamic';
import Link from 'next/link';
import truncate from 'lodash/truncate';
import {
  Checkbox, Icon, Modal, Popup
} from 'semantic-ui-react';

import VisuallyHidden from 'components/VisuallyHidden/VisuallyHidden';
import packageThumbnail from 'static/images/thumbnail_package.svg';
import DetailsPopup from '../DetailsPopup/DetailsPopup';
import { DashboardContext } from 'context/dashboardContext';

import './TeamProjectPrimaryCol.scss';

const GraphicPreview = dynamic( () => import(
  /* webpackChunkName: "graphicPreview" */
  'components/admin/GraphicPreview/GraphicPreview'
) );

const PackagePreview = dynamic( () => import(
  /* webpackChunkName: "packagePreview" */
  'components/admin/PackagePreview/PackagePreview'
) );

const ProjectPreviewContent = dynamic( () => import(
  /* webpackChunkName: "projectPreviewContent" */
  'components/admin/ProjectPreview/ProjectPreviewContent/ProjectPreviewContent'
) );

const handleDataActionsOffClick = e => {
  // Check if click target is a data actions menu link
  const isDataActionsMenuLink = e.target.classList.contains( 'projects_data_actions_action' );

  if ( isDataActionsMenuLink ) return;

  // If click target is not actions menu toggle button
  if ( !e.target.classList.contains( 'ellipsis' ) ) {
    // Close any open menus
    const openDataActionsMenu = document.querySelector( '.displayDataActions' );

    if ( openDataActionsMenu !== null ) openDataActionsMenu.classList.remove( 'displayDataActions' );

    // Remove the document click event handler since no open menus
    document.removeEventListener( 'click', handleDataActionsOffClick );
  }
};

const toggleDataActions = e => {
  e.persist();

  // Select target table td element
  const parentTD = e.target.closest( '.items_table_item' );

  if ( parentTD.classList.contains( 'displayDataActions' ) ) {
    parentTD.classList.remove( 'displayDataActions' );
  } else {
    // Close any other open dataActions menus
    const openDataActionsMenu = document.querySelector( '.displayDataActions' );

    if ( openDataActionsMenu !== null ) openDataActionsMenu.classList.remove( 'displayDataActions' );

    // Display target dataActions menu
    parentTD.classList.add( 'displayDataActions' );

    // Add a document click event listener to close any open data actions menus on 'off' clicks
    document.addEventListener( 'click', handleDataActionsOffClick );
  }
};

const TeamProjectPrimaryCol = ( {
  d,
  d: { id },
  header
} ) => {
  const { dispatch, state } = useContext( DashboardContext );

  const selectedItems = state?.selected?.selectedItems ? state.selected.selectedItems : new Map();

  const isDraft = d.status === 'DRAFT';
  const isPublishing = d.status === 'PUBLISHING';
  const actions = [
    'Edit', 'Preview', 'Files'
  ];
  const Trigger = isPublishing ? 'span' : 'a';
  const Title = isPublishing ? 'span' : Link;
  const projectTitleLength = d[header.name].length >= 35;

  const getThumbnailStyles = type => {
    switch ( type.toLowerCase() ) {
      case 'package':
        return { borderRadius: '10px' };

      default:
        return {};
    }
  };

  // todo share with my dashboard
  const getThumbnail = item => {
    const type = item?.__typename;
    let url = '';
    let alt = '';

    if ( item?.thumbnail?.signedUrl ) {
      url = item.thumbnail.signedUrl;
      alt = item.thumbnail.alt;
    } else if ( type?.toLowerCase() === 'package' ) {
      url = packageThumbnail;
      alt = 'Package';
    }

    if ( url ) {
      return (
        <img
          className={ isDraft ? 'draft' : null }
          style={ getThumbnailStyles( type ) }
          src={ url }
          alt={ alt }
        />
      );
    }

    return (
      <div className="placeholder outer">
        <div className="placeholder inner" />
      </div>
    );
  };

  const getEditUrl = ( format = '' ) => {
    if ( isPublishing || !format ) return null;

    // TEMP Package || DocumentFile url
    if ( d.__typename === 'Package' ) return `/admin/package/${id}`;
    if ( d.__typename === 'DocumentFile' ) return `/admin/document/${id}`;
    if ( d.__typename === 'GraphicProject' ) return `/admin/project/graphic/${id}`;

    if ( format === 'pretty' ) {
      if ( d.__typename === 'GraphicProject' ) return `/admin/project/graphic/${id}/edit`;

      return `/admin/project/video/${id}/edit`;
    }
    if ( format === 'long' ) {
      if ( d.__typename === 'GraphicProject' ) return `/admin/project?content=graphic&id=${id}&action=edit`;

      return `/admin/project?content=video&id=${id}&action=edit`;
    }
  };

  const getTitleCls = () => `projects_data_title${isPublishing ? ' isPublishing' : ''}`;
  const getActionCls = () => `linkStyle projects_data_actions_action${isPublishing ? ' isPublishing' : ''}`;

  // screenSize: Semantic UI modal size prop value
  const getPkgModalSize = screenSize => {
    if ( d.__typename === 'Package' ) {
      return { size: screenSize };
    }

    return {};
  };

  const getContentPreviewComponent = () => {
    switch ( d.__typename ) {
      case 'Package':
        return PackagePreview;
      case 'GraphicProject':
        return GraphicPreview;
      case 'VideoProject':
      default:
        return ProjectPreviewContent;
    }
  };

  const ContentPreview = getContentPreviewComponent();

  const toggle = ( _, data ) => {
    dispatch( { type: 'UPDATE_SELECTED', payload: { data, selected: selectedItems } } );
  };

  return (
    <Fragment>
      <div className="projects_actions">
        <Checkbox
          { ...( isPublishing ? {} : { 'data-label': id } ) }
          checked={ !!selectedItems.get( `${id}` ) }
          onChange={ toggle }
          disabled={ isPublishing }
        />
      </div>
      <div className="projects_thumbnail">
        <div className="wrapper">
          { getThumbnail( d ) }
          { isDraft && (
            <p className="draft-overlay" style={ getThumbnailStyles( d.__typename ) }>
              <span>{ d.status }</span>
            </p>
          ) }
        </div>
      </div>
      <div className="projects_data">
        { /**
         * Display tooltip for keyboard accessibility.
         * Tradeoff? Can't use <Link> in trigger and no
         * page transition animation, but err on side
         * of better accessibility.
         */ }
        { projectTitleLength ? (
          <Popup
            trigger={ (
              <Trigger href={ getEditUrl( 'pretty' ) } className={ getTitleCls() }>
                <span aria-hidden>{ truncate( d[header.name], { length: 35 } ) }</span>
                <VisuallyHidden el="span">{ d[header.name] }</VisuallyHidden>
              </Trigger>
            ) }
            content={ d[header.name] }
            hideOnScroll
            inverted
            on={ ['hover', 'focus'] }
            size="mini"
          />
        ) : (
          <Title as={ getEditUrl( 'pretty' ) } href={ getEditUrl( 'long' ) } prefetch={ false }>
            <Trigger className={ getTitleCls() }>{ d[header.name] }</Trigger>
          </Title>
        ) }
        <div className="projects_data_actions">
          <div className="projects_data_actions_wrapper">
            { isPublishing
              ? actions.map( ( action, i ) => (
                <Fragment key={ `${action}-${id}` }>
                  <span className={ getActionCls() }>{ action }</span>
                  { i < actions.length - 1 && <span className="separator">|</span> }
                </Fragment>
              ) )
              : (
                <Fragment>
                  <Link as={ getEditUrl( 'pretty' ) } href={ getEditUrl( 'long' ) } prefetch={ false }>
                    <a className={ getActionCls() }>Edit</a>
                  </Link>
                  <span className="separator">|</span>
                  <Modal
                    trigger={ (
                      <button type="button" className={ getActionCls() }>
                        Preview
                      </button>
                    ) }
                    closeIcon
                    className={ `${d.__typename}-preview` }
                    { ...getPkgModalSize( 'fullscreen' ) }
                  >
                    <Modal.Content>
                      <ContentPreview data={ d } id={ id } />
                    </Modal.Content>
                  </Modal>
                  { d.__typename !== 'DocumentFile' && (
                    <Fragment>
                      <span className="separator">|</span>
                      <DetailsPopup id={ id } team={ state.team } />
                    </Fragment>
                  ) }
                </Fragment>
              ) }
          </div>
          <button
            type="button"
            className="linkStyle projects_data_actions_mobileToggle"
            onClick={ toggleDataActions }
          >
            <Icon name="ellipsis vertical" />
          </button>
        </div>
      </div>
    </Fragment>
  );
};

TeamProjectPrimaryCol.propTypes = {
  d: PropTypes.object,
  header: PropTypes.object,
  selectedItems: PropTypes.instanceOf( Map ),
  toggleItemSelection: PropTypes.func,
  team: PropTypes.object
};

export default TeamProjectPrimaryCol;
