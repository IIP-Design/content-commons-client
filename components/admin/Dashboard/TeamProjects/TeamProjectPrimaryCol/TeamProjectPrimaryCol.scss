@import 'styles/colors.scss';

.projects {
  .isPublishing {
    cursor: default;
  }

  &_actions {
    margin-right: 15px;
    @media screen and (max-width: 767px) {
      margin-right: 0;
    }
  }

  &_thumbnail {
    flex: 1;
    margin-right: 15px;
    @media screen and (max-width: 767px) {
      margin-right: 0;
      padding: 0 3px;
      max-width: 66px;
    }

    > .wrapper {
      position: relative;

      img {
        display: block;
        width: 100%;

        &.draft {
          filter: blur(0.4px) grayscale(40%);
        }
      }

      > .draft-overlay {
        display: flex;
        align-items: center;
        justify-content: center;
        position: absolute;
        top: 0;
        right: 0;
        bottom: 0;
        left: 0;
        background-color: rgba(0, 0, 0, 0.4);

        > span {
          color: #fff;
          text-shadow: 0px 0px 4px #000;

          @media screen and (min-width: 768px) {
            font-size: 1rem;
          }
        }
      }

      > .placeholder {
        &.outer {
          position: relative;
          height: 0;
          padding-top: 56.25%;
          overflow: hidden;
          background-color: $light-grey;
        }

        &.inner {
          position: absolute;
          top: 0;
          left: 0;
          height: 100%;
          width: 100%;
        }
      }
    }
  }

  &_data {
    position: relative;
    flex: 2;
    display: flex;
    flex-direction: column;

    &_title {
      flex: 1;
      margin-bottom: 0;
      font-size: 18px;
      font-weight: bold;
      color: $blue;
      @media screen and (max-width: 767px) {
        max-width: 155px;
        font-size: 14px;
      }

      &:hover {
        color: $light-blue;
      }

      &.isPublishing {
        &:hover {
          color: inherit;
        }
      }
    }

    &_actions {
      button {
        -webkit-appearance: none;
        padding: 0;
        border: none;
        background-color: transparent;
      }

      &_wrapper {
        .separator {
          padding-left: 0.25rem;
          padding-right: 0.25rem;
        }

        @media screen and (max-width: 767px) {
          display: none;
          position: absolute;
          top: 0;
          right: 8px;
          z-index: 2;
          padding: 10px;
          background-color: #fff;
          box-shadow: 1px 1px 1px 1px $medium-grey;

          > span {
            display: none;
          }
        }
      }

      &_action {
        font-family: 'Source Sans Pro';
        font-size: 11px;
        font-weight: normal;
        @media screen and (max-width: 767px) {
          display: block;
          padding: 3px;
          margin-bottom: 8px;
          font-size: 1em;

          &:not(:first-child) {
            padding: 3px !important;
          }
        }
      }

      &_mobileToggle {
        position: absolute;
        top: 0;
        right: 5px;
        font-size: 1.5em;
        color: $blue;
        @media screen and (min-width: 768px) {
          display: none;
        }

        i.icon {
          margin: 0;
          width: auto;
        }
      }
    }
  }
}
