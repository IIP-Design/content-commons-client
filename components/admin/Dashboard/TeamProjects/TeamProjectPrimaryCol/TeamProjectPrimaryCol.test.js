import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import { Checkbox, Popup } from 'semantic-ui-react';
import TeamProjectPrimaryCol from './TeamProjectPrimaryCol';

jest.mock( 'next/dynamic', () => () => 'dynamically-imported' );
jest.mock( 'react', () => ( {
  ...jest.requireActual( 'react' ),
  useContext: () => ( { state: {
    selected: {
      selectedItems: new Map( [['999', true]] ),
    },
    team: 'Press Office',
  } } ),
} ) );

const props = {
  d: {
    id: '111',
    createdAt: 'May 9, 2019',
    updatedAt: 'May 9, 2019',
    projectTitle: 'Test Project',
    author: 'Jane Doe',
    status: 'PUBLISHED',
    visibility: 'PUBLIC',
    thumbnail: {
      alt: 'the alt text',
      url: 'https://website.com',
    },
  },
  header: {
    label: 'PROJECT TITLE',
    name: 'projectTitle',
  },
};

const Component = <TeamProjectPrimaryCol { ...props } />;

describe( '<TeamProjectPrimaryCol />', () => {
  it( 'renders without crashing', () => {
    const wrapper = shallow( Component );

    expect( wrapper.exists() ).toEqual( true );
    expect( toJSON( wrapper ) ).toMatchSnapshot();
  } );

  it( 'renders an unchecked Checkbox if item is *not* selected', () => {
    const wrapper = shallow( Component );
    const checkbox = wrapper.find( Checkbox );

    expect( checkbox.prop( 'checked' ) ).toEqual( false );
  } );

  it( 'renders a checked Checkbox if item is selected', () => {
    const wrapper = shallow( Component );

    wrapper.setProps( { selectedItems: new Map( [[props.d.id, true]] ) } );

    const checkbox = wrapper.find( Checkbox );

    expect( checkbox.prop( 'checked' ) ).toEqual( true );
  } );

  it( 'checking/unchecking the Checkbox calls toggleItemSelection', () => {
    const wrapper = shallow( Component );
    const checkbox = wrapper.find( Checkbox );

    checkbox.simulate( 'change' );
    expect( props.toggleItemSelection ).toHaveBeenCalled();
  } );

  it( 'does not render a Popup if project title is less than 35 characters', () => {
    const wrapper = shallow( Component );
    const popup = wrapper.find( Popup );

    expect( props.d.projectTitle.length ).toBeLessThanOrEqual( 35 );
    expect( popup.exists() ).toEqual( false );
  } );

  it( 'renders a Popup with a truncated project title if it is over 35 characters', () => {
    const wrapper = shallow( Component );

    wrapper.setProps( {
      d: {
        ...props.d,
        projectTitle: 'Test Project Test Project Test Project Test Project',
      },
    } );
    const popup = wrapper.find( Popup );

    expect( popup.exists() ).toEqual( true );
    expect( popup.prop( 'content' ) )
      .toEqual( 'Test Project Test Project Test Project Test Project' );
    expect( popup.prop( 'content' ).length ).toBeGreaterThan( 35 );
    expect( toJSON( popup ) ).toMatchSnapshot();
  } );

  it( 'renders "draft" class value for thumbnail img if status is DRAFT', () => {
    const wrapper = shallow( Component );
    const thumbnail = () => wrapper.find( '.projects_thumbnail img' );

    // initial value
    expect( thumbnail().prop( 'className' ) ).toEqual( null );

    wrapper.setProps( { d: { ...props.d, status: 'DRAFT' } } );
    expect( thumbnail().prop( 'className' ) ).toEqual( 'draft' );
  } );

  it( 'renders an overlay with "DRAFT" text for thumbnail img if status is DRAFT', () => {
    const wrapper = shallow( Component );
    const overlay = () => wrapper.find( '.draft-overlay' );

    // does not exist if status !== DRAFT
    expect( overlay().exists() ).toEqual( false );

    wrapper.setProps( { d: { ...props.d, status: 'DRAFT' } } );
    expect( overlay().exists() ).toEqual( true );
    expect( overlay().contains( <span>DRAFT</span> ) ).toEqual( true );
  } );

  it( 'renders a placeholder if there is no thumbnail url', () => {
    const wrapper = shallow( Component );
    const placeholder = () => wrapper.find( '.placeholder' );
    const innerPlaceholder = <div className="placeholder inner" />;

    // does not exist if there is a thumbnail url
    expect( placeholder().exists() ).toEqual( false );

    wrapper.setProps( {
      d: {
        ...props.d,
        thumbnail: { url: '', alt: '' },
      },
    } );
    expect( placeholder().exists() ).toEqual( true );
    expect( placeholder().contains( innerPlaceholder ) )
      .toEqual( true );
  } );

  it( 'renders spans for actions if status is PUBLISHING', () => {
    const wrapper = shallow( Component );

    wrapper.setProps( { d: { ...props.d, status: 'PUBLISHING' } } );

    const actions = [
      'Edit', 'Preview', 'Files',
    ];
    const spans = wrapper.find( '.projects_data_actions_action' );

    spans.forEach( ( span, i ) => {
      expect( span.name() ).toEqual( 'span' );
      expect( span.hasClass( 'isPublishing' ) ).toEqual( true );
      expect( span.text() ).toEqual( actions[i] );
    } );
  } );

  it( 'renders a span for the project title if status is PUBLISHING', () => {
    const wrapper = shallow( Component );

    wrapper.setProps( { d: { ...props.d, status: 'PUBLISHING' } } );

    const title = wrapper.find( '.projects_data_title' );

    expect( title.name() ).toEqual( 'span' );
    expect( title.hasClass( 'isPublishing' ) ).toEqual( true );
  } );

  it( 'renders a span for the Popup trigger if status is PUBLISHING & the project title is over 35 characters', () => {
    const wrapper = shallow( Component );

    wrapper.setProps( {
      d: {
        ...props.d,
        projectTitle: 'Test Project Test Project Test Project Test Project',
        status: 'PUBLISHING',
      },
    } );

    const popup = wrapper.find( Popup );

    expect( popup.prop( 'content' ).length ).toBeGreaterThan( 35 );
    expect( popup.prop( 'trigger' ).type ).toEqual( 'span' );
    expect( popup.prop( 'trigger' ).props.className )
      .toEqual( 'projects_data_title isPublishing' );
  } );

  it( 'renders a disabled checkbox if status is PUBLISHING', () => {
    const wrapper = shallow( Component );

    wrapper.setProps( { d: { ...props.d, status: 'PUBLISHING' } } );
    const checkbox = wrapper.find( Checkbox );

    expect( checkbox.prop( 'disabled' ) ).toEqual( true );
  } );

  it( 'does not render a data-label checkbox attribute if status is PUBLISHING', () => {
    const wrapper = shallow( Component );

    wrapper.setProps( { d: { ...props.d, status: 'PUBLISHING' } } );
    const checkbox = wrapper.find( Checkbox );
    const attrs = Object.keys( checkbox.props() );

    expect( attrs.includes( 'data-label' ) ).toEqual( false );
  } );
} );
