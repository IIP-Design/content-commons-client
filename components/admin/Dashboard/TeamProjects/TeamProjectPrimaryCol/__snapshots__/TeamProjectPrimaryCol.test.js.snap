// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TeamProjectPrimaryCol /> renders a Popup with a truncated project title if it is over 35 characters 1`] = `
<Popup
  content="Test Project Test Project Test Project Test Project"
  disabled={false}
  eventsEnabled={true}
  hideOnScroll={true}
  inverted={true}
  offset={0}
  on={
    Array [
      "hover",
      "focus",
    ]
  }
  pinned={false}
  position="top left"
  size="mini"
  trigger={
    <a
      className="projects_data_title"
      href="/admin/project/video/111/edit"
    >
      <span
        aria-hidden={true}
      >
        Test Project Test Project Test P...
      </span>
      <VisuallyHidden
        el="span"
      >
        Test Project Test Project Test Project Test Project
      </VisuallyHidden>
    </a>
  }
>
  <Portal
    closeOnDocumentClick={true}
    closeOnEscape={true}
    closeOnTriggerBlur={true}
    closeOnTriggerClick={false}
    closeOnTriggerMouseLeave={true}
    eventPool="default"
    mouseEnterDelay={50}
    mouseLeaveDelay={70}
    onClose={[Function]}
    onMount={[Function]}
    onOpen={[Function]}
    onUnmount={[Function]}
    openOnTriggerClick={false}
    openOnTriggerFocus={true}
    openOnTriggerMouseEnter={true}
    trigger={
      <a
        className="projects_data_title"
        href="/admin/project/video/111/edit"
      >
        <span
          aria-hidden={true}
        >
          Test Project Test Project Test P...
        </span>
        <VisuallyHidden
          el="span"
        >
          Test Project Test Project Test Project Test Project
        </VisuallyHidden>
      </a>
    }
    triggerRef={
      Object {
        "current": <a
          class="projects_data_title"
          href="/admin/project/video/111/edit"
        >
          <span
            aria-hidden="true"
          >
            Test Project Test Project Test P...
          </span>
          <span
            class="hide-visually"
          >
            Test Project Test Project Test Project Test Project
          </span>
        </a>,
      }
    }
  >
    <Ref
      innerRef={[Function]}
    >
      <RefFindNode
        innerRef={[Function]}
      >
        <a
          className="projects_data_title"
          href="/admin/project/video/111/edit"
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          <span
            aria-hidden={true}
          >
            Test Project Test Project Test P...
          </span>
          <VisuallyHidden
            el="span"
          >
            <span
              className="hide-visually"
            >
              Test Project Test Project Test Project Test Project
            </span>
          </VisuallyHidden>
        </a>
      </RefFindNode>
    </Ref>
  </Portal>
</Popup>
`;

exports[`<TeamProjectPrimaryCol /> renders without crashing 1`] = `
<TeamProjectPrimaryCol
  d={
    Object {
      "author": "Jane Doe",
      "createdAt": "May 9, 2019",
      "id": "111",
      "projectTitle": "Test Project",
      "status": "PUBLISHED",
      "thumbnail": Object {
        "alt": "the alt text",
        "url": "https://website.com",
      },
      "updatedAt": "May 9, 2019",
      "visibility": "PUBLIC",
    }
  }
  header={
    Object {
      "label": "PROJECT TITLE",
      "name": "projectTitle",
    }
  }
>
  <div
    className="projects_actions"
  >
    <Checkbox
      checked={false}
      data-label="111"
      disabled={false}
      onChange={[Function]}
      type="checkbox"
    >
      <div
        className="ui fitted checkbox"
        data-label="111"
        onChange={[Function]}
        onClick={[Function]}
        onMouseDown={[Function]}
        onMouseUp={[Function]}
      >
        <Ref
          innerRef={
            Object {
              "current": <input
                class="hidden"
                readonly=""
                tabindex="0"
                type="checkbox"
                value=""
              />,
            }
          }
        >
          <RefFindNode
            innerRef={
              Object {
                "current": <input
                  class="hidden"
                  readonly=""
                  tabindex="0"
                  type="checkbox"
                  value=""
                />,
              }
            }
          >
            <input
              checked={false}
              className="hidden"
              disabled={false}
              readOnly={true}
              tabIndex={0}
              type="checkbox"
            />
          </RefFindNode>
        </Ref>
        <Ref
          innerRef={
            Object {
              "current": <label />,
            }
          }
        >
          <RefFindNode
            innerRef={
              Object {
                "current": <label />,
              }
            }
          >
            <label />
          </RefFindNode>
        </Ref>
      </div>
    </Checkbox>
  </div>
  <div
    className="projects_thumbnail"
  >
    <div
      className="wrapper"
    >
      <div
        className="placeholder outer"
      >
        <div
          className="placeholder inner"
        />
      </div>
    </div>
  </div>
  <div
    className="projects_data"
  >
    <Link
      as="/admin/project/video/111/edit"
      href="/admin/project?content=video&id=111&action=edit"
      prefetch={false}
    >
      <a
        className="projects_data_title"
        href="/admin/project/video/111/edit"
        onClick={[Function]}
      >
        Test Project
      </a>
    </Link>
    <div
      className="projects_data_actions"
    >
      <div
        className="projects_data_actions_wrapper"
      >
        <Link
          as="/admin/project/video/111/edit"
          href="/admin/project?content=video&id=111&action=edit"
          prefetch={false}
        >
          <a
            className="linkStyle projects_data_actions_action"
            href="/admin/project/video/111/edit"
            onClick={[Function]}
          >
            Edit
          </a>
        </Link>
        <span
          className="separator"
        >
          |
        </span>
        <Modal
          centered={true}
          className="undefined-preview"
          closeIcon={true}
          closeOnDimmerClick={true}
          closeOnDocumentClick={false}
          dimmer={true}
          eventPool="Modal"
          trigger={
            <button
              className="linkStyle projects_data_actions_action"
              type="button"
            >
              Preview
            </button>
          }
        >
          <Portal
            closeOnDocumentClick={false}
            closeOnEscape={true}
            eventPool="Modal"
            mountNode={<body />}
            onClose={[Function]}
            onMount={[Function]}
            onOpen={[Function]}
            onUnmount={[Function]}
            openOnTriggerClick={true}
            trigger={
              <button
                className="linkStyle projects_data_actions_action"
                type="button"
              >
                Preview
              </button>
            }
          >
            <Ref
              innerRef={[Function]}
            >
              <RefFindNode
                innerRef={[Function]}
              >
                <button
                  className="linkStyle projects_data_actions_action"
                  onBlur={[Function]}
                  onClick={[Function]}
                  onFocus={[Function]}
                  onMouseEnter={[Function]}
                  onMouseLeave={[Function]}
                  type="button"
                >
                  Preview
                </button>
              </RefFindNode>
            </Ref>
          </Portal>
        </Modal>
        <span
          className="separator"
        >
          |
        </span>
        <details-popup
          id="111"
          team="Press Office"
        />
      </div>
      <button
        className="linkStyle projects_data_actions_mobileToggle"
        onClick={[Function]}
        type="button"
      >
        <Icon
          as="i"
          name="ellipsis vertical"
        >
          <i
            aria-hidden="true"
            className="ellipsis vertical icon"
            onClick={[Function]}
          />
        </Icon>
      </button>
    </div>
  </div>
</TeamProjectPrimaryCol>
`;
